sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationPropertySource.java

```diff
@@ -74,22 +76,29 @@ public Object getProperty(String name) {
         return properties.get(name);
     }
 
-    protected static String getLabelName(String[] labelFilter) {
-        if (labelFilter == null) {
+    protected static String getLabelName(String[] labelFilters) {
+        if (labelFilters == null) {
             return "";
         }
-        StringBuilder labelName = new StringBuilder();
-        for (String label : labelFilter) {
-
-            labelName.append((labelName.length() == 0) ? label : "," + label);
-        }
-        return labelName.toString();
+        return String.join(",", labelFilters);
     }
 
     List<ConfigurationSetting> getFeatureFlagSettings() {
         return featureConfigurationSettings;
     }
 
+    protected void processFeatureFlag(String key, FeatureFlagConfigurationSetting setting, List<String> trimStrings) {
```

Comment:
Let's move this back to the featureflag related classes. The abstract class should know nothing about how a feature flag is processed.

Reply:
If we want to reduce the duplication, the better way is to provide a static method with return value in the feature flag property source, and in both the snapshot one and feature flag one calls that method. 

Just like this pseudocode:

```java
protected Map<String, Object> processFeatureFlag(String key, FeatureFlagConfigurationSetting setting, List<String> trimStrings, ReplicaClient replicaClient) {


TracingInfo tracing = replicaClient.getTracingInfo();
        featureConfigurationSettings.add(setting);
        FeatureFlagConfigurationSetting featureFlag = setting;

        String configName = FEATURE_MANAGEMENT_KEY + setting.getKey().trim().substring(FEATURE_FLAG_PREFIX.length());

        updateTelemetry(featureFlag, tracing);

        return (configName, createFeature(featureFlag));
}

Reply:
Should this update be good enough?